Sources/SwiftKueryORM/Database.swift

```diff
@@ -65,16 +65,23 @@ public class Database {
     }
 
     /// Constructor for a custom connection generator
-    public init(generator: @escaping () -> Connection?) {
+    public init(generator: @escaping ((Connection?, QueryError?) -> ()) -> ()) {
         self.connectionStrategy = .generator(generator)
     }
 
     /// Connection getter: either new connection from pool
     /// or call the custom generator
-    public func getConnection() -> Connection? {
+    public func getConnection(task: @escaping (Connection?, QueryError?) -> ()) {
```

Comment:
We should update the documentation comments for `init()` and `getConnection()` to describe the parameters and give usage examples.